drivers: dac: esp32: avoid out-of-range channel ID

kartben opened the pull request:
fix channel ID check in dac_esp32_channel_setup as it was allowing to set up a channel with ID greater than the number of channels.

marekmatej reviewed (approved):
Please fix the commit message to use capitalize-case.

kartben commented:
> Please fix the commit message to use capitalize-case.

I am not sure what you mean? Sorry
Feel free to push the fix directly to my branch.

kartben committed:
drivers: dac: esp32: avoid out-of-range channel ID

Fix channel ID check in dac_esp32_channel_setup as it was allowing to
set up a channel with ID greater than the number of channels.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben commented:
> > > Please fix the commit message to use capitalize-case.
> > 
> > 
> > I am not sure what you mean? Sorry Feel free to push the fix directly to my branch.
> 
> "fix channel ID check" -> "Fix channel ID check"

Sure 🤷🏼
